File: src/lib/api/env.ts
import { invoke } from "@/lib/runtime/client/core";
import type { BackupInfo, EnvConflict } from "@/types/env";

export async function checkEnvConflicts(appType: string): Promise<EnvConflict[]> {
  return invoke("check_env_conflicts", { app: appType });
}

export async function deleteEnvVars(
  conflicts: EnvConflict[],
): Promise<BackupInfo> {
  return invoke("delete_env_vars", { conflicts });
}

export async function restoreEnvBackup(backupPath: string): Promise<void> {
  return invoke("restore_env_backup", { backupPath });
}

export async function checkAllEnvConflicts(): Promise<
  Record<string, EnvConflict[]>
> {
  const apps = ["claude", "codex", "gemini"];
  const results: Record<string, EnvConflict[]> = {};

  await Promise.all(
    apps.map(async (app) => {
      try {
        results[app] = await checkEnvConflicts(app);
      } catch (error) {
        console.error(`检查 ${app} 环境变量失败:`, error);
        results[app] = [];
      }
    }),
  );

  return results;
}

File: backend/src/commands/env.rs
use crate::services::env_checker::EnvConflict;
use crate::services::env_manager::BackupInfo;

pub(crate) fn check_env_conflicts_internal(app: String) -> Result<Vec<EnvConflict>, String> {
    crate::services::env_checker::check_env_conflicts(&app)
}

pub(crate) fn delete_env_vars_internal(conflicts: Vec<EnvConflict>) -> Result<BackupInfo, String> {
    crate::services::env_manager::delete_env_vars(conflicts)
}

pub(crate) fn restore_env_backup_internal(backup_path: String) -> Result<(), String> {
    crate::services::env_manager::restore_from_backup(backup_path)
}
